src/system/file_path.cc

```diff
@@ -16,123 +16,110 @@
+boost::filesystem::path FilePath::root_dir=boost::filesystem::path(".");
 
 FilePath::FilePath(string file_path, const  FileType ft)
 : file_type_(ft)
 {
-	abs_file_path_ = file_path;
-    if (output_dir == "") {
+    if (output_dir == boost::filesystem::path(".")) {
     	WarningOut() << "Creating FileName object before set_io_dirs is called. No file path resolving." << std::endl;
+    	abs_file_path_ = boost::filesystem::path(file_path);
         return;
     }
 
-    substitute_value();
+    substitute_value(file_path);
+	abs_file_path_ = boost::filesystem::path(file_path);
     if (ft == input_file) {
-    	if ( FilePath::is_absolute_path(abs_file_path_) ) {
+    	if ( abs_file_path_.is_absolute() ) {
     	} else {
-            abs_file_path_ = root_dir + DIR_DELIMITER + abs_file_path_;
+            abs_file_path_ = root_dir / file_path;
     	}
     } else if (ft == output_file) {
-        if ( FilePath::is_absolute_path(abs_file_path_) ) {
-            if (abs_file_path_.substr(0, output_dir.size()) == output_dir) {
-            	abs_file_path_=abs_file_path_.substr(output_dir.size()+1);
+        if ( abs_file_path_.is_absolute() ) {
+            if (abs_file_path_.string().substr(0, output_dir.string().size()) == output_dir.string()) {
+            	abs_file_path_ = boost::filesystem::path( abs_file_path_.string().substr(output_dir.string().size()+1) );
             } else {
-                THROW( ExcAbsOutputPath() << EI_Path( abs_file_path_ ) );
+                THROW( ExcAbsOutputPath() << EI_Path( abs_file_path_.string() ) );
             }
         }
-        abs_file_path_ = output_dir + DIR_DELIMITER + abs_file_path_;
+        abs_file_path_ = output_dir / abs_file_path_;
     }
 
 }
 
 
 
-void FilePath::set_io_dirs(const string working_dir, const string root_input_dir,const string input,const string output) {
+void FilePath::set_io_dirs(const string working_dir, const string root, const string input, const string output) {
+	ASSERT_EQ(working_dir, ".").error();
+    FilePath::set_dirs(root, input, output);
+}
+
+
+void FilePath::set_dirs(const string root, const string input, const string output) {
     // root directory
-    root_dir = root_input_dir;
-
-    // relative output dir is relative to working directory
-    // this is possibly independent of position of the main input file
-	output_dir = "";
-    if ( FilePath::is_absolute_path(output) ) {
-    	vector<string> dirs;
-    	boost::split(dirs, output,  boost::is_any_of("/"));
-    	for (vector<string>::iterator it = dirs.begin(); it != dirs.end(); ++it) {
-    	    if ( !(*it).size() ) continue;
-    	    if ( !output_dir.size() ) {
-#ifdef FLOW123D_HAVE_CYGWIN
-    	    	output_dir = (*it);
-#else
-    	    	output_dir = DIR_DELIMITER + *it;
-#endif // FLOW123D_HAVE_CYGWIN
-    	    } else {
-            	output_dir = output_dir + DIR_DELIMITER + *it;
-    	    }
-    	    FilePath::create_dir(output_dir);
-        }
+	root_dir = boost::filesystem::path(root);
+
```

Comment:
Make canonical absolute path for the root_dir here.
1. It is necessary to prevent warning in `FilePath` constructor if the root is ".",
2. Simplifies further code for setting `output_dir`